Can't build the project on ubuntu 14.04, gcc 4.8 with ComputeCpp CE 0.1.2

Dear all,

I try to build the project on Ubuntu 14.04, gcc 4.8 with ComputeCpp.

I passed the following flags to cmake in order to build the project:
`cmake -DCOMPUTECPP_PACKAGE_ROOT_DIR=/home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux -DOpenCL_INCLUDE_DIR=/usr/local/cuda/include -DPARALLEL_STL_BENCHMARKS=ON  ..` 

Cmake run without error. Then I run make, I get the following error:

> Scanning dependencies of target sycl_exclusive_scan_integration_header
[ 53%] Building ComputeCpp integration header file /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/build/benchmarks/sycl_exclusive_scan.cpp.sycl
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/benchmarks/sycl_exclusive_scan.cpp:36:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/execution_policy:48:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux/include/CL/sycl.hpp:1:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux/include/CL/../SYCL/sycl.hpp:20:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux/include/SYCL/sycl_interface.h:135:
/home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux/include/SYCL/apis.h:902:9: error: [Computecpp:CC0004]: a SYCL kernel with the name class
      ExclusiveScanAlgorithm1 (alias typename detail::enable_functor<class (lambda at
      /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/algorithm/buffer_algorithms.hpp:514:43), class
      ExclusiveScanAlgorithm1>::kernel_name) already exists
        typename detail::enable_functor<functorT, nameT>::kernel_name,
        ^
/home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.1.2-Linux/include/SYCL/apis.h:902:9: note: original SYCL kernel with name class
      ExclusiveScanAlgorithm1 (alias typename detail::enable_functor<class (lambda at
      /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/algorithm/buffer_algorithms.hpp:399:43), class
      ExclusiveScanAlgorithm1>::kernel_name) defined here
1 error generated.
make[2]: *** [benchmarks/sycl_exclusive_scan.cpp.sycl] Error 1
make[1]: *** [benchmarks/CMakeFiles/sycl_exclusive_scan_integration_header.dir/all] Error 2
make: *** [all] Error 2

Hi dear all, now I use computecpp latest realise (ComputeCpp-CE-0.2.1-Linux), but I get the same errors:

[ 53%] Building ComputeCpp integration header file /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/build/benchmarks/sycl_exclusive_scan.cpp.sycl
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/benchmarks/sycl_exclusive_scan.cpp:36:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/execution_policy:48:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.2.1-Linux/include/CL/sycl.hpp:1:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.2.1-Linux/include/CL/../SYCL/sycl.hpp:20:
In file included from /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.2.1-Linux/include/SYCL/sycl_interface.h:139:
/home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.2.1-Linux/include/SYCL/apis.h:969:9: error: [Computecpp:CC0004]: a SYCL kernel with the name class
      ExclusiveScanAlgorithm1 (alias typename detail::enable_functor<class (lambda at
      /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/algorithm/buffer_algorithms.hpp:514:43), class
      ExclusiveScanAlgorithm1>::kernel_name) already exists
        typename detail::enable_functor<functorT, nameT>::kernel_name,
        ^
/home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/ComputeCpp-CE-0.2.1-Linux/include/SYCL/apis.h:969:9: note: original SYCL kernel with name class
      ExclusiveScanAlgorithm1 (alias typename detail::enable_functor<class (lambda at
      /home/rokiatou/Documents/Thèse/sycl_kronos/different_implementation_of_sycl/SyclParallelSTL/include/sycl/algorithm/buffer_algorithms.hpp:399:43), class
      ExclusiveScanAlgorithm1>::kernel_name) defined here
1 error generated.
make[2]: *** [benchmarks/sycl_exclusive_scan.cpp.sycl] Error 1
make[1]: *** [benchmarks/CMakeFiles/sycl_exclusive_scan_integration_header.dir/all] Error 2
make: *** [all] Error 2


Can anyone help me please?
Thanks.

--- a/include/sycl/algorithm/buffer_algorithms.hpp
+++ b/include/sycl/algorithm/buffer_algorithms.hpp
@@ -30,6 +30,7 @@
 #define __SYCL_IMPL_BUFFER_ALGORITHM__
 
 #include <sycl/helpers/sycl_buffers.hpp>
+#include <sycl/helpers/sycl_namegen.hpp>
 
 namespace sycl {
 namespace impl {
@@ -287,7 +288,8 @@ B buffer_map2reduce(ExecutionPolicy &snp,
     cl::sycl::accessor<B, 1, cl::sycl::access::mode::read_write,
                        cl::sycl::access::target::local>
       sum { cl::sycl::range<1>(d.nb_work_item), cgh };
-    cgh.parallel_for_work_group<typename ExecutionPolicy::kernelName>(rng, [=](cl::sycl::group<1> grp) {
+    cgh.parallel_for_work_group<typename ExecutionPolicy::kernelName>(
+        cl::sycl::range<1>{rng.get_global()}, [=](cl::sycl::group<1> grp) {
       size_t group_id = grp.get(0);
       //assert(group_id < d.nb_work_group);
       size_t group_begin = group_id * d.size_per_work_group;
@@ -395,7 +397,8 @@ void buffer_mapscan(ExecutionPolicy &snp,
                        cl::sycl::access::target::local>
       scratch { cl::sycl::range<1> { d.size_per_work_group }, cgh };
 
-    cgh.parallel_for_work_group<typename ExecutionPolicy::kernelName>(rng_wg, rng_wi,
+
+    cgh.parallel_for_work_group<cl::sycl::helpers::NameGen<0, typename ExecutionPolicy::kernelName> >(rng_wg, rng_wi,
                                           [=](cl::sycl::group<1> grp) {
       size_t group_id = grp.get(0);
       //assert(group_id < d.nb_work_group);
@@ -509,8 +512,7 @@ void buffer_mapscan(ExecutionPolicy &snp,
       <cl::sycl::access::mode::read_write>(cgh);
     auto read_scan = scan.template get_access
       <cl::sycl::access::mode::read>(cgh);
-
-    cgh.parallel_for_work_group<typename ExecutionPolicy::kernelName>(rng_wg, rng_wi,
+    cgh.parallel_for_work_group<cl::sycl::helpers::NameGen<1, typename ExecutionPolicy::kernelName>>(rng_wg, rng_wi,
                                           [=](cl::sycl::group<1> grp) {
       size_t group_id = grp.get(0);
       B acc = read_scan[group_id];

--- a/include/sycl/algorithm/inner_product.hpp
+++ b/include/sycl/algorithm/inner_product.hpp
@@ -177,7 +177,7 @@ T inner_product(ExecutionPolicy &snp, InputIt1 first1, InputIt1 last1,
   auto input_buff1 = sycl::helpers::make_const_buffer(first1, last1);
   auto input_buff2 = sycl::helpers::make_const_buffer(first2, last2);
 
-  auto map = [&](size_t pos, value_type_1 x, value_type_2 y) {
+  auto map = [=](size_t pos, value_type_1 x, value_type_2 y) {
     return op2(x, y);
   };
 

